smslibrary/src/main/java/com/eis/smslibrary/SMSDeliveredBroadcastReceiver.java

```diff
@@ -9,32 +9,41 @@
 
 import com.eis.smslibrary.listeners.SMSDeliveredListener;
 
+import java.util.ArrayList;
+import java.util.Collections;
+
 /**
  * Broadcast receiver for delivered messages, called by Android Library.
  * Must be instantiated and set as receiver with context.registerReceiver(...).
  * There has to be one different DeliveredBroadcastReceiver per message sent,
  * so every IntentFilter name has to be different
  *
- * @author Marco Cognolato
+ * @author Marco Cognolato, Giovanni Velludo
  */
 public class SMSDeliveredBroadcastReceiver extends BroadcastReceiver {
-    private SMSDeliveredListener listener;
-    private SMSMessage message;
+
+    private final SMSDeliveredListener listener;
+    private final ArrayList<SMSPart> messageParts;
+    private final SMSPeer peer;
 
     /**
      * Constructor for the custom {@link BroadcastReceiver}.
      *
-     * @param message  that will be sent.
-     * @param listener to be called when the operation is completed.
+     * @param parts    parts of the message to be delivered.
+     * @param listener listener to be called when the operation is completed
+     * @param peer     peer to whom the message will be delivered.
```

Comment:
definite/indefinite articles are missing

Reply:
What indefinite articles are missing?

Reply:
the parts
the listener
the peer
definite/indefinite is because in general you can choose according to your needs

Reply:
Ok